pyat/developers.rst

```diff
@@ -114,24 +120,77 @@ To upload a release to PyPI, you will need to be a 'maintainer' of
 For testing any version that you have installed, the simple snippet in
 ``README.rst`` is sufficient.
 
-* Decide the Python versions that should be supported in the release
-   * Set these Python versions in ``python_requires`` in ``setup.cfg``
-   * Set at least these Python versions as ``python-version`` in ``.github/workflows/python-tests.yml``
-* Determine the minimum Numpy version that is required for those Python versions
-   * Set this numpy version in ``install_requires`` in ``setup.cfg``
-* Push a tag ``pyat-x.y.z`` to Github
+Decide the Python versions that should be supported in the release
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Set these Python versions in ``requires-python`` in ``pyproject.toml``
+* Set at least these Python versions as ``python-version`` in ``.github/workflows/python-tests.yml``
+
+Determine the minimum ``numpy`` and ``scipy`` versions:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* the version ensuring the requirements necessary to **run** PyAT is set in the
+  ``dependencies`` item of the ``[project]`` section of ``pyproject.toml``
+* The version required to **build** PyAT is set in the ``requires`` item of the
+  ``[build]`` section of ``pyproject.toml``. It depends on the python version and
+  must be higher or equal to the "run" version.
+* To avoid version conflicts with the user's existing libraries, the pre-compiled
+  binaries are built with the exact minimum library versions. This ensures that the
+  user's libraries are more recent than the one AT has been compiled with. For
+  that, a copy of ``pyproject.toml`` named ``githubproject.toml`` is used for
+  compilation. In this copy, the numpy version specifications are set using strict
+  equality(``==``) instead of minimum (``>=``). Apart from these lines, the 2 files
+  should be strictly identical.
+
+Prepare the "Release notes"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A draft can be obtained by creating a new tag on GitHub. Click "Draft a new release"
+in the releases page, choose a new tag like "pyat-x.y.z", select
+the master branch as target. In the description area, choose the present release in
+the "previous tag" pull-down, and press "Generate release notes".
+
+The generated notes can now be copied and edited. You can then either cancel or
+save the release as a draft while editing the release notes.
+
+The ``## What's changed`` section should be splitted in ``## Bug fixes`` and
```

Comment:
'splitted in' -> 'split into'

Are you suggesting having two top-level titles ('Bug fixes' and 'New features') only, or keeping What's Changed?